### PR TITLE
Tag docker container from semver git tags

### PR DESCRIPTION
**Pull request by aethr**
## Description

Ensure docker build step can extract the right tags when the triggering event was a tag in format `vX.Y.Z` was pushed.

See: https://github.com/docker/metadata-action?tab=readme-ov-file#semver

## Checklist

Please confirm that this pull request has done the following:

- [ ] Tests added
- [ ] Documentation added (where applicable)

### CODE REVIEW

**Comment by lewisjared** [comment]
For OpenMethane, we used `type=pep440,pattern={{version}}`. 

[PEP440](https://peps.python.org/pep-0440/) is Python the specification for version identifiers accepted by Python. Typically semver is followed so semver could be seen as a subset of PEP440, but there might be some minor differences in how prereleases are encoded. For our purposes we should just make them consistent and move on.

@mikapfl Do you have anything to add as a TLDR of PEP440 vs semver?

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -52,6 +52,8 @@ jobs:
             type=schedule
             type=ref,event=branch
             type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
```
Comment (lewisjared) [inline .github/workflows/build.yaml:56]:
```suggestion
            type=pep440,pattern={{version}}
            type=pep440,pattern={{major}}.{{minor}}
```

**Review by lewisjared — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yaml (hunk 1 above)
Comment (aethr) [reply .github/workflows/build.yaml]: Thanks, this is really helpful. After reading your comment I just copied what we have in https://github.com/openmethane/openmethane/blob/main/.github/workflows/build_docker.yaml#L53C1-L53C44, do you think that's suitable?

Comment (lewisjared) [reply .github/workflows/build.yaml]: Yes. It has worked as expected when combined with the CI release process

**Review by lewisjared — APPROVED** [review]

**Comment by mikapfl** [comment]
The big difference between PEP440 and semver (as far as I understand) is that pep440 mainly talks about the format of version numbers, while semver talks about their meaning. Of course, semver also prescribes a special format (which is pep440-compliant) and pep440 also has some ideas about meaning (e.g. release candidates and development releases), but generally, pep440 doesn't have notions of "breaking changes" or "api compatibility" or similar things encoded in the version number, while semver has.

**Pull request merged**